docs/ocis/adr/0021-service-authorization.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "21. Service authorization"
+date: 2023-01-18T16:07:00+01:00
+weight: 21
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0020-service-authorization.md
+---
+
+* Status: proposed
+* Deciders: @butonic
+* Date: 2023-01-18
+
+## Context and Problem Statement
+
+Some services need access to file content without a user being logged in. We currently pass the owner or manager
+of a space in events which allows the search service to impersonate that user to extract metadata from the changed resource.
+There are two problems with this:
+1. The service could get all permissions of the user and gain write permission
+2. There is a race condition where the user in the event might no longer have read permission, causing the index to go stale
+
+The race condition will become more of an issue when we start working on a workflow engine.
+
+How can we grant services the least amount of permissions required for their purpose?
+
+## Decision Drivers
+
+* It should be possible to represent this as servicePrincipals in the libregraph API, similar to the [MS Graph servicePrincipal](https://learn.microsoft.com/en-us/graph/api/resources/serviceprincipal?view=graph-rest-1.0).
+* Services should check permissions using the ocis permissions or reva auth service, we don't want to introduce a new mechanism for this
+
+## Considered Options
+
+* [Service Accounts](#service-accounts)
+* [Auth Manager type Space-Owner](#auth-manager-type-space-owner)
```

Comment:
go micro may have a concept for service accounts: https://github.com/go-micro/go-micro/blob/master/auth/auth.go#L55-L68 and uses [Rules](https://github.com/go-micro/go-micro/blob/master/auth/auth.go#L55-L68) to limit access to resources, see also the [example](https://github.com/go-micro/examples/blob/main/auth/auth.go)